caffe2/utils/conversions.h

```diff
@@ -0,0 +1,185 @@
+#pragma once
+
+#include <caffe2/core/types.h>
+
+#ifdef __CUDA_ARCH__
+#include <cuda_fp16.h>
+#endif
+
+#ifdef __CUDA_ARCH__
+#define CONVERSIONS_DECL __host__ __device__ inline
+#else
+#define CONVERSIONS_DECL inline
+#endif
+
+namespace caffe2 {
+
+namespace convert {
+
+namespace {
+inline float16 cpu_float2half_rn(float f) {
+    float16 ret;
+
+    unsigned x = *((int*)(void*)(&f));
```

Comment:
Unfortunately both `*((int*)(void*)(&f))` and *(reinterpret_cast<unsigned int*>(&f)) will break builds with -fstrict-aliasing (error: dereferencing type-punned pointer will break strict-aliasing rules)
To fix that we can just use the pointer itself: 
`unsigned x* = reinterpret_cast<unsigned int*>(&f);`
 and in line 101 as well.